### PR TITLE
Configs: handy functions to prepend and append; more validation

### PR DESCRIPTION
**Pull request by vladsavelyev**
* Add `append_config_paths` and `prepend_config_paths` in addition to `set_config_paths`
* Into `_validate_configs`, added some actual validation: parsing of all tomls and reporting errors

### CODE REVIEW

File: cpg_utils/config.py [hunk 1]
```diff
@@ -4,29 +4,29 @@
...
+from cpg_utils import to_path
 
 # We use these globals for lazy initialization, but pylint doesn't like that.
 # pylint: disable=global-statement, invalid-name
 _config_paths = _val.split(',') if (_val := os.getenv('CPG_CONFIG_PATH')) else []
 _config: Optional[frozendict] = None  # Cached config, initialized lazily.
 
 
-def _validate_configs(config_paths: List[str]) -> None:
+def _validate_configs(config_paths: list[str]) -> None:
```
Comment (MattWellie) [inline cpg_utils/config.py:17]: I know this isn't the right PR for this comment, but I'd like this validation method to do some more validation 😬 

Any thoughts on whether we should attempt to load each identified file using `toml` to check that they both exist and can be parsed correctly?

File: cpg_utils/config.py [hunk 2]
```diff
@@ -45,6 +45,29 @@ def set_config_paths(config_paths: List[str]) -> None:
         _config = None  # Make sure the config gets reloaded.
 
 
+def prepend_config_paths(config_paths: list[str]) -> None:
+    """
+    Prepend the list of config paths. Equivalent to `dict.set_defaults`:
+    values in current CPG_CONFIG_PATH will have the precedence over the provided
+    `config_paths` when merging configs.
+    """
+    if _env_var := os.environ.get('CPG_CONFIG_PATH'):
```
Comment (MattWellie) [inline cpg_utils/config.py:70]: love a cheeky walrus

**Review by MattWellie — APPROVED** [review]

File: cpg_utils/config.py (hunk 1 above)
Comment (vladsavelyev) [reply cpg_utils/config.py:17]: Yeah, good idea. Loading and parsing from remote might slow down the application a bit, but is probably worth it

Comment (vladsavelyev) [reply cpg_utils/config.py:17]: I added some validation into that function

**Pull request merged**